<h3>Data set #2:<br>
Resident population at the beginning of the year (by gender)<br>
http://osp.stat.gov.lt/en/statistiniu-rodikliu-analize?portletFormName=visualization&hash=e62988b7-de7a-4c72-83b3-9cad28f4f250
</h3>

In [154]:
#pulling csv data from file
import csv
filename='Source_datasets/Exported matrix table_d2.csv'
with open(filename, 'r') as f:
    data = list(csv.reader(f))
#print (data)
#print (f)
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['Total by age', 'Republic of Lithuania', 'Males', '2016', '1329607']
['', '', 'Females', '2016', '1558951']
['', 'Alytus county', 'Males', '2016', '67486']
['', '', 'Females', '2016', '77623']
['', 'Alytus t. mun.', 'Males', '2016', '24896']
['', '', 'Females', '2016', '29535']
['', 'Alytus d. mun.', 'Males', '2016', '13080']
['', '', 'Females', '2016', '13751']
['', 'Druskininkai mun.', 'Males', '2016', '8938']
['', '', 'Females', '2016', '11462']
['', 'Lazdijai d. mun.', 'Males', '2016', '9731']
['', '', 'Females', '2016', '10707']
['', 'Varėna d. mun.', 'Males', '2016', '10841']
['', '', 'Females', '2016', '12168']
['', 'Kaunas county', 'Males', '2016', '263124']
['', '', 'Females', '2016', '314234']
['', 'Birštonas mun.', 'Males', '2016', '1923']
['', '', 'Females', '2016', '2407']
['', 'Jonava d. mun.', 'Males', '2016', '19977']
['', '', 'Females', '2016', '23219']
['', 'Kaišiadorys d. mun.', 'Ma

In [155]:
#remove aggregated rows, as aggregation will take place later e.g. in pivotjs
keysList=['Republic','county']
tobedeletedList=[]
ageCurser={}#to tag age layers 
updated_lengh=len(data)
for row_num in range(0,updated_lengh):
    for key in keysList:
        #print (key)
        if key in data[row_num][1]:
            tobedeletedList.append(row_num)#for males row
            tobedeletedList.append(row_num+1)#for the females row
            if 'Republic' in key:
                ageCurser[data[row_num][0]]=row_num
                
import operator

sorted_ageCurser = sorted(ageCurser.items(), key=operator.itemgetter(1))

print (tobedeletedList)
print (ageCurser)
print('sorted cursor:{}'.format(sorted_ageCurser))


for aggregated_row_num in tobedeletedList:
    #print (data[aggregated_row_num])
    data[aggregated_row_num]=['']
    #tobedeletedList[:] = [x - 1 for x in tobedeletedList]
   #print (tobedeletedList)
    
for row_num in range(0,len(data)):
    print (data[row_num])


[1, 2, 3, 4, 15, 16, 33, 34, 49, 50, 63, 64, 77, 78, 93, 94, 103, 104, 113, 114, 127, 128, 145, 146, 147, 148, 159, 160, 177, 178, 193, 194, 207, 208, 221, 222, 237, 238, 247, 248, 257, 258, 271, 272, 289, 290, 291, 292, 303, 304, 321, 322, 337, 338, 351, 352, 365, 366, 381, 382, 391, 392, 401, 402, 415, 416, 433, 434, 435, 436, 447, 448, 465, 466, 481, 482, 495, 496, 509, 510, 525, 526, 535, 536, 545, 546, 559, 560]
{'Working-age population': 289, 'Total by age': 1, 'Population (0–15 years)': 145, 'Pension age population': 433}
sorted cursor:[('Total by age', 1), ('Population (0–15 years)', 145), ('Working-age population', 289), ('Pension age population', 433)]
['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['']
['']
['']
['']
['', 'Alytus t. mun.', 'Males', '2016', '24896']
['', '', 'Females', '2016', '29535']
['', 'Alytus d. mun.', 'Males', '2016', '13080']
['', '', 'Females', '2016', '13751']
['', 'Druskininkai mun.', 'Males', '2016', '8938']
[

In [156]:
#remove first layer as it is aggregated (total by age)
totalbyAgeLayer_start_end=[sorted_ageCurser[0][1],sorted_ageCurser[1][1]]
print (totalbyAgeLayer_start_end)

del_curser=totalbyAgeLayer_start_end[1]-1 #-1 is important not to delete the start of the next layer
print (del_curser)

for i in range(totalbyAgeLayer_start_end[0],totalbyAgeLayer_start_end[1]):
    print ("delete row {}:{}".format(del_curser,data[del_curser]))
    data[del_curser]=['']
    del_curser=del_curser-1
    
for row_num in range(0,len(data)):
    print (data[row_num])



[1, 145]
144
delete row 144:['', '', 'Females', '2016', '49407']
delete row 143:['', 'Vilnius d. mun.', 'Males', '2016', '45782']
delete row 142:['', '', 'Females', '2016', '300439']
delete row 141:['', 'Vilnius c. mun.', 'Males', '2016', '243054']
delete row 140:['', '', 'Females', '2016', '19526']
delete row 139:['', 'Ukmergė d. mun.', 'Males', '2016', '16634']
delete row 138:['', '', 'Females', '2016', '17483']
delete row 137:['', 'Trakai d. mun.', 'Males', '2016', '15537']
delete row 136:['', '', 'Females', '2016', '13295']
delete row 135:['', 'Švenčionys d. mun.', 'Males', '2016', '11885']
delete row 134:['', '', 'Females', '2016', '8468']
delete row 133:['', 'Širvintos d. mun.', 'Males', '2016', '7606']
delete row 132:['', '', 'Females', '2016', '16896']
delete row 131:['', 'Šalčininkai d. mun.', 'Males', '2016', '15409']
delete row 130:['', '', 'Females', '2016', '12614']
delete row 129:['', 'Elektrėnai mun.', 'Males', '2016', '11345']
delete row 128:['']
delete row 127:['']
del

<p>sorted cursor:[('Total by age', 1), ('Population (0–15 years)', 145), ('Working-age population', 289), ('Pension age population', 433)]</p>

In [157]:
#add missing Age tag layers values
firstAgeLayer_curser=[sorted_ageCurser[1][1],sorted_ageCurser[2][1]]
firstAgeLayer_tag=sorted_ageCurser[1][0]
secondAgeLayer_curser=[sorted_ageCurser[2][1],sorted_ageCurser[3][1]]
secondAgeLayer_tag=sorted_ageCurser[2][0]
thirdAgeLayer_curser=[sorted_ageCurser[3][1],len(data)]
thirdAgeLayer_tag=sorted_ageCurser[3][0]

print(firstAgeLayer_curser)
print(secondAgeLayer_curser)
print(thirdAgeLayer_curser)
print(firstAgeLayer_tag)

for i in range(firstAgeLayer_curser[0],firstAgeLayer_curser[1]):
    data[i][0]=firstAgeLayer_tag

for i in range(secondAgeLayer_curser[0],secondAgeLayer_curser[1]):
    data[i][0]=secondAgeLayer_tag

for i in range(thirdAgeLayer_curser[0],thirdAgeLayer_curser[1]):
    data[i][0]=thirdAgeLayer_tag
    
for row_num in range(0,len(data)):
    print (data[row_num])

[145, 289]
[289, 433]
[433, 577]
Population (0–15 years)
['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['Population (0–15 years)']
['Population (0–15 years)']
['Population (0–15 years)']
['Population (0–15 years)']
['Population (0–15 years)'

In [158]:
#permanent clean of unwanted rows in dataset
tobedeletedList=[]
for row_num in range (1,len(data)):
    if (len(data[row_num])<3):#regarding temporary header size
        #print ("row num{}:{}".format(i,data[row_num]))
        tobedeletedList.append(row_num)
print (tobedeletedList)
for aggregated_row_num in tobedeletedList:
    #print (data[aggregated_row_num])
    del data[aggregated_row_num]
    tobedeletedList[:] = [x - 1 for x in tobedeletedList]
   #print (tobedeletedList)
    
for row_num in range(0,len(data)):
    print (data[row_num])

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 159, 160, 177, 178, 193, 194, 207, 208, 221, 222, 237, 238, 247, 248, 257, 258, 271, 272, 289, 290, 291, 292, 303, 304, 321, 322, 337, 338, 351, 352, 365, 366, 381, 382, 391, 392, 401, 402, 415, 416, 433, 434, 435, 436, 447, 448, 465, 466, 481, 482, 495, 496, 509, 510, 525, 526, 535, 536, 545, 546, 559, 560]
['\ufeff', '', '', '', 'Resident population at the beginn

In [159]:
#add missing country values
for row_num in range(1,len(data)):
    data[row_num].append('Republic of Lithuania')
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['Population (0–15 years)', 'Alytus t. mun.', 'Males', '2016', '3924', 'Republic of Lithuania']
['Population (0–15 years)', '', 'Females', '2016', '3783', 'Republic of Lithuania']
['Population (0–15 years)', 'Alytus d. mun.', 'Males', '2016', '1941', 'Republic of Lithuania']
['Population (0–15 years)', '', 'Females', '2016', '1865', 'Republic of Lithuania']
['Population (0–15 years)', 'Druskininkai mun.', 'Males', '2016', '1350', 'Republic of Lithuania']
['Population (0–15 years)', '', 'Females', '2016', '1364', 'Republic of Lithuania']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Males', '2016', '1434', 'Republic of Lithuania']
['Population (0–15 years)', '', 'Females', '2016', '1448', 'Republic of Lithuania']
['Population (0–15 years)', 'Varėna d. mun.', 'Males', '2016', '1554', 'Republic of Lithuania']
['Population (0–15 years)', '', 'Females', '2016', '1480', 'Republic of Lithuania']
['Populati

In [160]:
#getting new county ranges (old dif*2+new)
def update_range(old_start,old_end,new_start):
    new_end_start=new_start+(old_end-old_start)*2
    return new_end_start

print (update_range(54,62,107))



123


In [161]:
#add missing county values
chuncks=[0,122,244]

for chunck in range(0,3):
    for row_num in range(1+chuncks[chunck],11+chuncks[chunck]):
            data[row_num].append('Alytus county')
    for row_num in range(11+chuncks[chunck],27+chuncks[chunck]):
            data[row_num].append('Kaunas county')
    for row_num in range(27+chuncks[chunck],41+chuncks[chunck]):
            data[row_num].append('Klaipeda county')
    for row_num in range(41+chuncks[chunck],53+chuncks[chunck]):
            data[row_num].append('Marijampole county')
    for row_num in range(53+chuncks[chunck],65+chuncks[chunck]):
            data[row_num].append('Panevezys county')
    for row_num in range(65+chuncks[chunck],79+chuncks[chunck]):
            data[row_num].append('Siauliai county')
    for row_num in range(79+chuncks[chunck],87+chuncks[chunck]):
            data[row_num].append('Taurage county')
    for row_num in range(87+chuncks[chunck],95+chuncks[chunck]):
            data[row_num].append('Telsiai county')
    for row_num in range(95+chuncks[chunck],107+chuncks[chunck]):
            data[row_num].append('Utena county')
    for row_num in range(107+chuncks[chunck],123+chuncks[chunck]):
            data[row_num].append('Vilnius county')

        
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['Population (0–15 years)', 'Alytus t. mun.', 'Males', '2016', '3924', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', '', 'Females', '2016', '3783', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus d. mun.', 'Males', '2016', '1941', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', '', 'Females', '2016', '1865', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Druskininkai mun.', 'Males', '2016', '1350', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', '', 'Females', '2016', '1364', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Males', '2016', '1434', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', '', 'Females', '2016', '1448', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Varėna d. mun.', 'Males',

In [162]:
#Adding missing municapility value for the Females rows
for row_num in range(1,len(data)):
    if not data[row_num][1]:
        #print (data[row_num])
        data[row_num][1]=data[row_num-1][1]
        #print (data[row_num])
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', '', 'Resident population at the beginning of the year | persons']
['Population (0–15 years)', 'Alytus t. mun.', 'Males', '2016', '3924', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus t. mun.', 'Females', '2016', '3783', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus d. mun.', 'Males', '2016', '1941', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus d. mun.', 'Females', '2016', '1865', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Druskininkai mun.', 'Males', '2016', '1350', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Druskininkai mun.', 'Females', '2016', '1364', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Males', '2016', '1434', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Females', '2016', '1448', 'Republic of Lithuania', 'Alytus co

In [163]:
#Adding collumn headers istead of the melformed header
headerList=['Age','Municipality','Sex','Time_period','Resident_population_at_the_beginning_of_the_year_persons','Country','County']
del data[0]
data.insert( 0, headerList)
for row_num in range(0,len(data)):
    print (data[row_num])

['Age', 'Municipality', 'Sex', 'Time_period', 'Resident_population_at_the_beginning_of_the_year_persons', 'Country', 'County']
['Population (0–15 years)', 'Alytus t. mun.', 'Males', '2016', '3924', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus t. mun.', 'Females', '2016', '3783', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus d. mun.', 'Males', '2016', '1941', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Alytus d. mun.', 'Females', '2016', '1865', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Druskininkai mun.', 'Males', '2016', '1350', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Druskininkai mun.', 'Females', '2016', '1364', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Males', '2016', '1434', 'Republic of Lithuania', 'Alytus county']
['Population (0–15 years)', 'Lazdijai d. mun.', 'Females', '2016', '

In [164]:
import csv
#print (data)
with open("Ready_datasets/readyDataSet2.csv", "w") as f:
    writer = csv.writer(f)
    writer.writerows(data)

<h4>Construct query</h4>
<p>
RPBYmunicipality_Query=PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n\ <br>
PREFIX OGI: <http://ogi.eu/#>\n\<br>
PREFIX qb: <http://purl.org/linked-data/cube#>\n\<br>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n\<br>
CONSTRUCT{\n\<br>
		?observation a qb:Observation;\n\<br>
				OGI:age ?Age;\n\<br>
				OGI:municipality ?Municipality;\n\<br>
				OGI:sex ?Sex;\n\<br>
				OGI:time_period ?Time_period;\n\<br>
				OGI:resident_population_at_the_beginning_of_the_year_persons<br> ?Resident_population_at_the_beginning_of_the_year_persons;\n\<br>
				OGI:counrty ?Country;\n\<br>
				OGI:county ?County;\n\<br>
				qb:dataSet OGI:RPBYmunicipality_ds;\n\<br>
				.\n\<br>
	}\n\<br>
WHERE {\n\<br>
BIND (uri(CONCAT('http://ogi.eu/#observations_',StrUUID())) AS ?observation).\n\<br>
}\n\<br>
</p>